pagefaultgames/pokerogue
# [Localization] Use gender context for dialogues (all 3 files)

## What are the changes the user will see?

Effectively none

## Why am I making these changes?

Right now we have the inconvenient situation of having `-male.json` and `-female.json` files for all dialogues.
I wanted to reduce the redundancy by merging those by using the [i18n `context`](https://www.i18next.com/translation-function/context) key.

## What are the changes from a developer perspective?

Since `en` is the single source of truth for the language keys I had to add a `"<key>_female": null` wherever there was a deviation in any language.

<sub>This change might also create new translation "voids" in [GitLocalize](https://gitlocalize.com/repo/9839). That's unfortunately a necessary evil</sub>

To be able to merge `-male` and `-female` dialogue files I used the file-compare tool of my IDE. All Asian languages had no differences between m & f.

### Screenshots/Videos

<sup>The `MALE:` and `FEMALE:` were just added in the dialogue files for this video. It is not part of the PR changes! This is just to make it clear that it was loading the gender specific key</sup>

https://github.com/user-attachments/assets/9825af1e-58a9-4440-ab2a-108cf2483dee

https://github.com/user-attachments/assets/fdf9a287-46c1-4274-bdb1-0515144fe2ea


## How to test the changes?
- Start the game in a language that has gender specific achievment names
    - e.g. `de`, `es`, `pt_BR`, `fr`
- Play runs until you encounter trainers. Check the `json` file if the messages differ by gender.
- Change your gender to a different one
- Check them again => Should render the gender context string

## Checklist
